document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -430,52 +422,53 @@ public boolean bulkUpsert(Map<Key, Document> documents) {
   }
 
   private int[] bulkUpsertImpl(Map<Key, Document> documents) throws SQLException, IOException {
-    PreparedStatement preparedStatement = client
-        .prepareStatement(getUpsertSQL(), Statement.RETURN_GENERATED_KEYS);
-    for (Map.Entry<Key, Document> entry : documents.entrySet()) {
+    try (PreparedStatement preparedStatement = client
+            .prepareStatement(getUpsertSQL(), Statement.RETURN_GENERATED_KEYS)) {
+      for (Map.Entry<Key, Document> entry : documents.entrySet()) {
 
-      Key key = entry.getKey();
-      String jsonString = prepareUpsertDocument(key, entry.getValue());
+        Key key = entry.getKey();
+        String jsonString = prepareUpsertDocument(key, entry.getValue());
 
-      preparedStatement.setString(1, key.toString());
-      preparedStatement.setString(2, jsonString);
-      preparedStatement.setString(3, jsonString);
+        preparedStatement.setString(1, key.toString());
+        preparedStatement.setString(2, jsonString);
+        preparedStatement.setString(3, jsonString);
 
-      preparedStatement.addBatch();
-    }
+        preparedStatement.addBatch();
+      }
 
-    return preparedStatement.executeBatch();
+      return preparedStatement.executeBatch();
+    }
   }
 
   @Override
   public Iterator<Document> bulkUpsertAndReturnOlderDocuments(Map<Key, Document> documents) throws IOException {
+    String query = null;
     try {
       String collect = documents.keySet().stream()
-          .map(val -> "'" + val.toString() + "'")
-          .collect(Collectors.joining(", "));
+              .map(val -> "'" + val.toString() + "'")
```

Comment:
nit: The new indentation seems like is 8 spaces but we follow Google's Java coding style and only 4 space when the statement extends onto the new line.